kernel/include/nano_internal.h

```diff
@@ -82,6 +82,56 @@ static inline unsigned int _Swap(unsigned int key)
 
 #ifdef CONFIG_USERSPACE
 /**
+ * @brief get the maximum number of free regions for memory domain partitions
+ *
+ * @return Max number of free regions
+ */
+extern u8_t _arch_get_max_partitions(void);
```

Comment:
No point in making a return value a single byte.  Just define it as an int, the code won't be any different and the interface will be more general.  It's not completely inconceivable that a future derived OS might support more than 255 separately-protected memory domains.  Desktop linux apps do this every day (three separate mmap's per shared library, dozens for the heap, etc...), it's not even a weird use case.

Reply:
yes good idea